Code review TDSClient

Milos-Curcic-MS opened the pull request.

Milos-Curcic-MS committed:
Added TDSClient for in depth connectivity testing

Milos-Curcic-MS committed:
Pre code-review commit

Milos-Curcic-MS committed:
Directory bug fix

Milos-Curcic-MS committed:
Library fix

Milos-Curcic-MS committed:
Library download bug fix

Milos-Curcic-MS committed:
Library bug fix 2.0

Milos-Curcic-MS committed:
Added admin elevation for advanced con tests

Milos-Curcic-MS committed:
Library file bug fix

Milos-Curcic-MS committed:
Bug fixes

Milos-Curcic-MS committed:
Library bug fix

Milos-Curcic-MS committed:
Library not being updated fix

Milos-Curcic-MS committed:
Minor bug fixes

Milos-Curcic-MS committed:
Moved conn policy script to another process

Milos-Curcic-MS committed:
Script location bug fixes

Milos-Curcic-MS committed:
Start-Job filePath bug fix

Milos-Curcic-MS committed:
Final commit

Milos-Curcic-MS committed:
Removed unnecessary output

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/signed)](https://cla.opensource.microsoft.com/Azure/SQL-Connectivity-Checker?pullRequest=2) <br/>All CLA requirements met.

Milos-Curcic-MS committed:
Added additional encryption information

Milos-Curcic-MS committed:
Always trusting remote certificates

Milos-Curcic-MS committed:
Tls version negotiation fix

Milos-Curcic-MS committed:
Library fix

Milos-Curcic-MS committed:
Moved to .NET Standard 2.1, added additional encryption data

Milos-Curcic-MS committed:
Returned to .NET Standard 2.0

Milos-Curcic-MS committed:
Added TLS 1.2 support and additional encryption output

Milos-Curcic-MS committed:
Added encryption protocol optional parameter

Milos-Curcic-MS committed:
Tls 1.3 not supported

Milos-Curcic-MS committed:
TLS defaults to the version decided by the operating system
Exception output fix

Milos-Curcic-MS committed:
Updated TDSClient library to .NET Standard 2.1

Milos-Curcic-MS committed:
Fixed exception output

Milos-Curcic-MS committed:
Added logging timestamp to TDSClient, reverted back to .NET Standard 2.0

Milos-Curcic-MS committed:
TLS default version fix

vitomaz-msft committed:
Update AdvancedConnectivityPolicyTests.ps1

Invoke-WebRequest was failing due to TLS.

vitomaz-msft committed:
bump to v1.0

- Removed tests based on SubnetCIDR
+ Added dummy parameters for User and Password
+ Set master as default database
+ Added CollectNetworkTrace option
+ Log and network trace are now collected to a folder and that folder is added to AllFiles.zip so customers can share them easily.

vitomaz-msft committed:
Some style improvements in TDSClient

vitomaz-msft committed:
Fix AzureSQLConnectivityCheckerResults temp folder creation

Milos-Curcic-MS committed:
Added execution policy check
Changed logging timezone to UTC

nimiljko commented on TDSClient/TDSClient/TDS/Client/TDSClientVersion.cs:
```diff
@@ -0,0 +1,49 @@
+        public bool Unpack(MemoryStream stream)
+        {
+            BuildNumber = BigEndianUtilities.ReadUShort(stream);
+            Minor = (byte)stream.ReadByte();
```
Why cast to byte here?

nimiljko commented on TDSClient/TDSClient/TDS/Client/TDSSQLTestClient.cs:
```diff
@@ -0,0 +1,223 @@
+            tdsMessageBody.AddOption("ServerName", (ushort)ServerName.Length, ServerName);
+            tdsMessageBody.AddOption("Password", (ushort)Password.Length, Password);
+            tdsMessageBody.AddOption("Database", (ushort)Database.Length, Database);
+            tdsMessageBody.AddOption("IntName", (ushort)"TDSSQLTestClient".Length, "TDSSQLTestClient");
```
Create "TDSSQLTestClient" as a constant, its already repeated twice in this line.

Milos-Curcic-MS committed:
Pre-merge commit
